### DISCUSSION TITLE
What is the right way within a script to measure the content of a node tree?

### DISCUSSION THREAD

**Discussion by ctrlcctrlv** [sile-typesetter/sile#1599] [Q&A] [answered]
I see there's a `question` tag, so I hope it's allowed to ask one.

I hope it will return a `SILE.measurement`.

**Comment by alerque** [comment, marked as answer]
If you just have a node, say an *hbox* or a *vbox* or something you should just be able to ask it for the measurement you want. All nodes have width and length properties that are measurements and (at least once they are shaped) should tell you about their sizes. An exact use case might shed some light on how/when to grab that info.

You might look at the definition of `float()` in `packages/frametricks.lua` for an example that also is an example of an "undo" (see #925):

```
local hbox = SILE.call("hbox", {}, content)
table.remove(SILE.typesetter.state.nodes) -- steal it back
```

That typeset a tree of content into an hbox, kept the result in a variable but nuked it from the actual output queue so it wouldn't appear anywhere. It then goes on to use `hbox.weight` and `hbox.width`.
